api/storage_service.cc

```diff
@@ -900,6 +902,129 @@ void set_storage_service(http_context& ctx, routes& r) {
             return make_ready_future<json::json_return_type>(map_to_key_value(std::move(status), res));
         });
     });
+
+    ss::sstable_info.set(r, [&ctx] (std::unique_ptr<request> req) {
+        auto ks = api::req_param<sstring>(*req, "keyspace", {}).value;
+        auto cf = api::req_param<sstring>(*req, "table", {}).value;
+
+        using table_sstables_list = std::vector<ss::table_sstables>;
+
+        return do_with(table_sstables_list{}, [ks, cf, &ctx](table_sstables_list& dst) {
+            return service::get_local_storage_service().db().map_reduce([&dst](table_sstables_list&& res) {
+                for (auto&& t : res) {
```

Comment:
The same issue from before, it's an iteration without preemption if it's not too much work, we can keep it as is, if it can potentially take a few ms we should make it safer to prevent stalls.

Reply:
What do you mean "safer"? Explicit yields? The map-reduce op should be yielding in itself really...

Reply:
right, but the for loop will not, so that's the question if the for loop can take anything more than a 1ms we should probably do something else

Reply:
I think it should be fairly safe. It will be a lot of string copying, but nothing really heavy. Plus, it is not a command you should run willy-nilly polling the server. If we find a case where it causes a stall, I can add yielding behaviour then. Hopefully it will be when we have coroutines. :-)